gui/src/app/pages/HomePage/AnalysisPyWindow/AnalysisPyWindow.tsx

```diff
@@ -0,0 +1,196 @@
+      const chainIds: number[] = [];
+      for (let i = 0; i < numChains; i++) {
+        for (let j = 0; j < numDrawsPerChain; j++) {
+          chainIds.push(i + 1);
+        }
+      }
+      return {
+        spData: {
+          sampling: {
+            draws: transpose(draws),
+            paramNames,
+            numChains,
+            chainIds,
+          },
+        },
+        scriptHeader: "",
+      };
+    } else {
+      return {
+        spData: {},
+        scriptHeader: getScriptHeaderForEmptyDraws(),
+      };
+    }
+  }, [draws, paramNames, numChains]);
+  return (
+    <Splitter
+      direction="vertical"
+      width={width}
+      height={height}
+      initialPosition={(3 * height) / 5}
+    >
+      <AnalysisPyFileEditor
+        width={0}
+        height={0}
+        fileName="analysis.py"
+        fileContent={data.analysisPyFileContent}
+        editedFileContent={data.ephemera.analysisPyFileContent}
+        setEditedFileContent={(content) => {
+          update({
+            type: "editFile",
+            content,
+            filename: ProjectKnownFiles.ANALYSISPYFILE,
+          });
+        }}
+        onSaveContent={() => {
+          update({
+            type: "commitFile",
+            filename: ProjectKnownFiles.ANALYSISPYFILE,
+          });
+        }}
+        consoleOutputDiv={consoleOutputDiv}
+        imageOutputDiv={imageOutputDiv}
+        readOnly={false}
+        spData={spData}
+        scriptHeader={scriptHeader}
+      />
+      <ConsoleOutputWindow
+        width={0}
+        height={0}
+        onDivElement={setConsoleOutputDiv}
+      />
+    </Splitter>
+  );
+};
+
+type ConsoleOutputWindowProps = {
+  width: number;
+  height: number;
+  onDivElement: (div: HTMLDivElement) => void;
+};
+
+const ConsoleOutputWindow: FunctionComponent<ConsoleOutputWindowProps> = ({
+  width,
+  height,
+  onDivElement,
+}) => {
+  return (
+    <div
+      style={{ position: "absolute", width, height, overflowY: "auto" }}
+      ref={onDivElement}
+    />
+  );
+};
+
+type ImageOutputWindowProps = {
+  width: number;
+  height: number;
+  onDivElement: (div: HTMLDivElement) => void;
+};
+
+const ImageOutputWindow: FunctionComponent<ImageOutputWindowProps> = ({
+  width,
+  height,
+  onDivElement,
+}) => {
+  return (
+    <div
+      style={{ position: "absolute", width, height, overflowY: "auto" }}
+      ref={onDivElement}
+    />
+  );
+};
+
+type RightPaneProps = {
+  width: number;
+  height: number;
+  onImageOutputDiv: (div: HTMLDivElement) => void;
+};
+
+const RightPane: FunctionComponent<RightPaneProps> = ({
+  width,
+  height,
+  onImageOutputDiv,
+}) => {
+  return (
+    <ImageOutputWindow
+      width={width}
+      height={height}
+      onDivElement={onImageOutputDiv}
+    />
+  );
+};
+
+const getScriptHeaderForEmptyDraws = () => {
+  return `
+raise Exception("You must run the sampler before executing the analysis script.")
+`;
+};
```

Comment:
Is this necessary if we're going to block the UI?

Reply:
You're right, I haven't blocked the run button in the UI yet. I'll do that now.

Reply:
The UI now does not allow running when there are no samples, and there's a helpful message on the editor.